### ISSUE TITLE
1.0-beta: regression with valid `LinkTo` vs `link-to` invocation patterns

### ISSUE THREAD

**Issue by kpfefferle** [typed-ember/glint#480]
Since Ember's own `<LinkTo>` component won't accept `undefined` passed to `@model` when the `@route` doesn't require a model, we have an internal component that does the following to conditionally apply the `@model` argument while otherwise sharing the same styling regardless:

```hbs
{{#let (if @model (component "link-to" model=@model) (component "link-to")) as |LinkComponent|}}
  <LinkComponent
    ...attributes
    @activeClass="custom-active-classes"
    @route={{@route}}
    class="custom-classes"
  >
    {{yield}}
  </LinkComponent>
{{/let}}
```

This works fine in earlier versions of Glint, but in the 1.0-beta it fails with the following error:

```
Argument of type '{ [NamedArgs]: true; activeClass: string; route: string; }' is not assignable to parameter of type 'string'.
```

My guess is this has something to do with some of the internal type refactoring? 🤔

**Comment by dfreeman** [comment]
This is a bug in the way we converted the types for `link-to` to account for our internal refactoring. The relationship between `LinkTo` and `link-to` is a troubling one for us, as the subtle differences between the two forms are a little tricky to account for, but we should definitely make sure this works.

**Comment by kpfefferle** [comment]
Confirmed fixed for us in `1.0.0-beta.2` - thanks @dfreeman!
